homeassistant/components/emoncms/coordinator.py

```diff
@@ -0,0 +1,31 @@
+"""DataUpdateCoordinator for the emoncms integration."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pyemoncms import EmoncmsClient
+
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class EmoncmsCoordinator(DataUpdateCoordinator[list[dict[str, Any]] | None]):
+    """Emoncms Data Update Coordinator."""
+
+    def __init__(
+        self, hass: HomeAssistant, emoncms_client: EmoncmsClient, config: ConfigType
```

Comment:
Let's only make the scan interval a parameter instead of the whole config

Reply:
Done